esmvaltool/diag_scripts/seaice/seaice_tsline.ncl

```diff
@@ -0,0 +1,338 @@
+; #############################################################################
+; LINE PLOTS OF SEA ICE AREA AND EXTENT vs TIME
+; Author: Daniel Senftleben (DLR, Germany)
+; ESMVal project
+; #############################################################################
+;
+; Description
+;   Creates line plot for sea ice area and extent versus time.
+;
+; Required diag_script_info attributes (diagnostics specific)
+;   month: selected month or annual mean.
+;   region: region to be plotted ( "Arctic" or "Antarctic").
+;
+; Optional diag_script_info attributes (diagnostics specific)
+;   fill_pole_hole: fill observational hole at North pole, Default: False
+;   styleset: specify line and color style
+;   EMs_in_lg: create legend label for each individual ensemble member
+;              (Default: "True")
+;   xy_line_legend: False: do not draw a legend inside the plot
+;   legend_outside: True: draw legend in an extra plot
+;
+; Required variable_info attributes (variable specific)
+;
+; Optional variable_info attributes (variable specific)
+;
+; Caveats
+;   CMOR units expected: areacello in m2, sic in %
+;
+; Modification history
+;   20190227-A_senf_da: Added provenance.
+;   20190218-A_senf_da: Ported to ESMValTool v2.0a1
+;   20180620-A_senf_da: Ported to ESMValTool v2
+;   20160906-A_senf_da: Added option to fill polar observation hole
+;   20151027-A_laue_ax: Moved call to 'write_references' to the beginning
+;                       of the code
+;   20150325-A_laue_ax: Modified reference tags used for acknowledgements
+;                       (projects, observations, etc)
+;   20140312-A_righ_ma: Adapted to new time_operations structure.
+;   20140129-A_senf_da: Modularized to fit to new structure.
+;   20130419-A_gott_kl: Implemented into ./diag_scripts.
+;   20130405-A_gott_kl: Implemented.
+;
+; #############################################################################
+load "$NCARG_ROOT/lib/ncarg/nclscripts/csm/shea_util.ncl"
+
+load "$diag_scripts/../interface_scripts/interface.ncl"
+load "$diag_scripts/../interface_scripts/logging.ncl"
+
+load "$diag_scripts/seaice/seaice_aux.ncl"
+
+load "$diag_scripts/shared/latlon.ncl"
+load "$diag_scripts/shared/set_operators.ncl"
+load "$diag_scripts/shared/statistics.ncl"
+
+load "$diag_scripts/shared/plot/style.ncl"
+load "$diag_scripts/shared/plot/aux_plotting.ncl"
+load "$diag_scripts/shared/plot/xy_line.ncl"
+
+
+begin
+
+  enter_msg(DIAG_SCRIPT, "")
+
+  n_variables = ListCount(variable_info)
+  var0 = variable_info[0]@short_name
+  field_type0 = variable_info[0]@field
+
+end
+
+
+begin
+
+  ; Get metadata items
+  info = select_metadata_by_name(input_file_info, var0)
+  n_datasets = ListCount(info)
+
+  ; Check field type
+  plot_fields = (/"T2Ms"/)
+  ind_f = any(field_type0.eq.plot_fields)
+  if (all(ismissing(ind_f))) then
+    error_msg("f", DIAG_SCRIPT, "", "can't use field " + field_type0)
+  end if
```

Comment:
See above. If you want to keep the check you can use the variable dimensions. Or simply remove it.